src/select/select-component.js

```diff
@@ -531,17 +531,33 @@ class Select extends React.Component<PropsT, SelectStateT> {
     this.setValue(valueArray.concat(item), item, STATE_CHANGE_TYPE.select);
   };
 
-  popValue = () => {
-    if (this.props.multi) {
-      const valueArray = [...this.props.value];
-      const valueLength = valueArray.length;
-      if (!valueLength) return;
-      if (valueArray[valueLength - 1].clearableValue === false) return;
-      const item = valueArray.pop();
-      this.setValue(valueArray, item, STATE_CHANGE_TYPE.remove);
+  backspaceValue = () => {
+    const item = this.popValue();
+    if (!item) {
+      return;
+    }
+    const valueLength = this.props.value.length;
+    const renderLabel = this.props.getValueLabel || this.getValueLabel;
+    const labelForInput = renderLabel({option: item, index: valueLength - 1});
+    // label might not be a string, it might be a Node of another kind.
+    if (typeof labelForInput === 'string') {
+      const remainingInput = labelForInput.slice(0, -1);
```

Comment:
Can the `remainingInput` value be configurable based on the `backspaceRemoves` prop?

If `backspaceRemoves === true` `remainingInput` should be an empty string, completely reseting the input value
otherwise, pop the last character like you have here.

Reply:
I looked into this a bit more as I was implementing the fix and found that `backspaceRemoves` actually toggles whether the backspace key can remove items or not. The change in behavior you proposed changes the meaning of `backspaceRemoves` and would be a breaking change in behavior, since it would mean that true/false values would always allow backspace key to remove the item.

Reply:
Good point. It does not make sense to repurpose the existing flag. I was concerned with how multi-select currently works with `backspaceRemoves`, which does not repopulate the input value. This PR does change default behavior in that way. To make for a completely non-breaking change we will need to include a new prop to control this interaction and default it to clearing the value.

Reply:
Yep, this goes to whether the behavior is expected or if it's unexpected to decide that.
If we're not sure, a new option might be good (if that's not considered too heavy-handed). `backspaceRemovesCharacter`?

Reply:
I tend to have a higher tolerance for making small behavior changes than others so I'm personally fine with landing, especially given that this seems like a preferable alternative. If it's requested, I can make a follow up to add the config. It's important to note that this change has come as a result of customer feedback to improve component UX and we believe this default behavior will make for better applications.

Reply:
Great point here. I'll update the description.